apps/verticalhome/js/pin_app_navigation.js

```diff
@@ -0,0 +1,294 @@
+'use strict';
+/* global ApplicationSource */
+/* global BookmarkSource */
+/* global CollectionSource */
+/* global configurator */
+/* global dispatchEvent */
+/* global GaiaGrid */
+
+(function(exports) {
+
+
+  /**
+   * This is PinAppNavigation's constructor.
+   * @param {Object} list [DOM container that contains pin elements]
+   */
+  function PinAppNavigation(list) {
+    if (!list && typeof list !== 'object') {
+      console.log('List of pin apps must be an object');
+      return;
+    }
+
+    this.elemList = list;
+    window.removeEventListener('keydown', this);
+    window.addEventListener('keydown', this);
+  };
+
+
+  PinAppNavigation.prototype = {
+
+    selector: '',
+    elements: null,
+    preventDef: false,
+    selectedElemIndex: 0,
+    elemList: null,
+    prevVerticalKey: null,
+
+    middleElem: false,
+
+
+    /**
+     * This function reset all pinns to default values.
+     * @return {[nothing]} [nothing]
+     */
+    reset: function reset() {
+      MoreAppsNavigation.clearNavigationEvents();
+      this.elements.sort(function(elem1, elem2) {
+        if (!elem1.dataset.index || !elem2.dataset.index) {
+          return 1;
+        }
+        return elem1.dataset.index - elem2.dataset.index;
+      });
```

Comment:
Why sort these on reset instead of when you initialize this.elements or add items to it.  Does this array ever get unsorted?